src/rapids_singlecell/preprocessing/_neighbors.py

```diff
@@ -262,6 +262,36 @@ def _get_connectivities(
     return connectivities
 
 
+def _trimming(cnts: cp_sparse.csr_matrix, trim: int) -> cp_sparse.csr_matrix:
+    from ._kernels._bbknn import cut_smaller_func, find_top_k_per_row_kernel
+
+    n_rows = cnts.shape[0]
+    vals_gpu = cp.zeros(n_rows, dtype=cp.float32)
+
+    threads_per_block = 64
+    blocks_per_grid = (n_rows + threads_per_block - 1) // threads_per_block
+
+    shared_mem_per_thread = trim * cp.dtype(cp.float32).itemsize
+    shared_mem_size = threads_per_block * shared_mem_per_thread
+
+    find_top_k_per_row_kernel(
+        (blocks_per_grid,),
+        (threads_per_block,),
+        (cnts.data, cnts.indptr, cnts.shape[0], trim, vals_gpu),
+        shared_mem=shared_mem_size,
+    )
+
+    for _ in range(2):
```

Comment:
?

Reply:
ah, I missed the `cnts = cnts.T.tocsr()` at the end.

so you transpose and convert to csr twice, I see.